### ISSUE TITLE
Update user profile

### ISSUE THREAD

**Issue by borutlatecheckout** [GetStream/stream-chat-swiftui#456]
## What are you trying to achieve?
I want to update user profile and see changes in the channel afterwards.

## If possible, how can you achieve this currently?
We are updating profile on the BE atm and once this is done, we restart the chat. But there is a slight delay after which we get updated info to be reflected in the channel.

## What would be the better way?
I would like to know what is the best solution for this scenario?
Is there maybe a function in SDK to update current user profile?

## GetStream Environment
**GetStream Chat version:** latest
**GetStream Chat frameworks:** StreamChat, StreamChatUI
**iOS version:** 17
**Swift version:** 5.10
**Xcode version:** 15.x
**Device:** iPhone

**Comment by borutlatecheckout** [comment]
This is most likely true, but cannot confirm for UIKit. In swiftUI this is certainly not updating for me. I would say that there is an issue with the binding. But would like to get a feedback from @martinmitrevski. Thanks @nuno-vieira for you assistance.

**Comment by martinmitrevski** [comment]
@borutlatecheckout the `viewModel.userFirstName` and the `ChatViewFactory` is something we don't have in the SDK, should be in your app. Can you share how are you using it and when do you update that property?

**Comment by borutlatecheckout** [comment]
You can ignore `userFirstName` since it is only used for the system chat message. For all other messages we read name and picture from the `message.author`.

ChatViewFactory looks like this:
```swift
final class ChatViewFactory: ViewFactory {
  typealias ActionHandler = GenericHandler<Action>
  @Injected(\.chatClient) public var chatClient
  var name: String?
  var action: ActionHandler?

  // custom message view
  func makeMessageContainerView(
    channel: ChatChannel, message: ChatMessage, width: CGFloat?, showsAllInfo: Bool,
    isInThread: Bool, scrolledId: Binding<String?>, quotedMessage: Binding<ChatMessage?>,
    onLongPress: @escaping (MessageDisplayInfo) -> Void, isLast: Bool
  ) -> some View {
    ChatMessageView(
      message: message,
      name: name ?? "User",
      showsAllInfo: showsAllInfo,
      action: action
    )
  }
}
```

And this is ChatMessageView:
```swift
struct ChatMessageView: View {
  let message: ChatMessage
  /// Name is only used for the initial/welcome message created on the BE.
  let name: String
  let showsAllInfo: Bool
  var action: ActionHandler?
  
  var body: some View {
    if message.isWelcome {
      ChatWelcomeView(date: message.createdAt, name: name)
    } else {
      regularMessage
    }
  }
}
```

I would assume I'm doing all things correctly. If I add a reaction, the message will update. So I suppose name and picture should change as well when a update profile.

**Comment by martinmitrevski** [comment]
Thanks, was able to reproduce this. We will look into it and let you know when we have an update.

**Comment by borutlatecheckout** [comment]
@martinmitrevski any update on this one?

**Comment by martinmitrevski** [comment]
Hey @borutlatecheckout, not yet - we have higher priority items at the moment.

**Comment by borutlatecheckout** [comment]
This issue is also present when I add url to message. `message.linkAttachments` is empty until I reload the chat.

**Comment by martinmitrevski** [comment]
Hey @borutlatecheckout,

This is fixed in 4.59.0. Thanks for your patience.

Best, Martin
